tests/mssqltestutils.py

```diff
@@ -90,29 +92,79 @@ def getTempPath(*args):
     return  os.path.abspath(tempPath)
 
 def create_test_db():
-    client = create_mssql_cli_client()
+    """
+    Creates database for test, using various status checks and retry logic for reliability.
+    - Calls helper method to check status of create db task, if possible
+    - Exits on successful response or retry period exceeds time limit
+    """
+    options = create_mssql_cli_options(database='master')
+    client = create_mssql_cli_client(options)
+
     local_machine_name = socket.gethostname().replace(
         '-', '_').replace('.', '_')
+
     test_db_name = u'mssqlcli_testdb_{0}_{1}'.format(
         local_machine_name, random_str())
     query_db_create = u"CREATE DATABASE {0};".format(test_db_name)
-    count = 0
-
-    # retry logic in case db create fails
-    while count < 5:
-        for _, _, status, _, is_create_error in client.execute_query(query_db_create):
-            if not is_create_error:
-                shutdown(client)
-                return test_db_name
-            # log warning to console
+
+    for _, _, status, _, is_create_error in client.execute_query(query_db_create):
+        if is_create_error:
+            # break loop to assert db creation failure
             warnings.warn('Test DB create failed with error: {0}'.format(status))
-        count += 1
+            break
+
+        if _is_client_db_on_cloud(client):
+            # retry logic is only supported for sql azure
+            create_db_status = _check_create_db_status(test_db_name, client)
+
+            if create_db_status == 'FAILED':
+                # break loop to assert db creation failure
+                break
+
+        shutdown(client)
+        return test_db_name
+
     shutdown(client)
 
     # cleanup db just in case, then raise exception
     clean_up_test_db(test_db_name)
     raise AssertionError("DB creation failed.")
```

Comment:
Can we put the failing exception in the assert message - this way we don't have look at logs